### ISSUE TITLE
Pass testsuite name to test case

### ISSUE THREAD

**Issue by tomasfejfar** [sebastianbergmann/phpunit#4468]
## Why?

There is TestCase::getName() that returns current test method name. It seems unreasonable that there are no such methods for suite and test case (though that can be obtained from `get_class($this)`). 

## How?

Test suite should be aware about it's name, so it can pass it to testCase when it's instantiated. Test suite could be null if the test is not ran as part of a test suite... 

### Steps to reproduce:

* there is a test case "IntegrationTest"
* this test case pulls configuration from ENV and based on it it tests different backend adapters with the same code
* it's a blackbox test against a closed system
* there are 3 different test suites running in parallel
* we send build id and $this->getName() in useragent when calling the API
* an exception occurs

### Expected
* I am able to tell from which of the three test suites caused the exception from the closed system's error log

### Actual 
* I am not able to get current test suite name, so I can't add it to the log context, so it cannot be propagated to the closed system's log

There is a Stackoverflow question related https://stackoverflow.com/q/25033345/112000

**Comment by sebastianbergmann** [comment]
I disagree that a test case should know the name of the test suite it is part of. The problem you have should be solved outside of PHPUnit, for instance by organizging your tests differently.

**Comment by tomasfejfar** [comment]
Care to elaborate more? Based on your comment I wonder why are there testsuites in the first place? Don't they also solve a problem that should be solved outside of PHPUnit?

**Comment by tomasfejfar** [comment]
Or maybe let me rephrase the question:

What would be a good way to add a logged that would have suite name in it's context?

**Comment by epdenouden** [comment]
@tomasfejfar @sebastianbergmann I'll answer this question at home later, this also came up with the dependency and TestDox refactoring

**Comment by epdenouden** [comment]
@tomasfejfar Thanks for taking the time to describe your use case in detail. Setups like this one are increasingly common thanks to hosted test automation.

@sebastianbergmann is right that PHPUnit doesn't offer a (documented!) interface for this and would best be solved from the 'outside': test collection structure by using groups and keeping track what runs where on the infra. Myself I have been gradually refactoring the internals to no longer _need_ `::getName()` except for pretty logfiles.

Tomáš are you interested in a videocall with screenshare to discuss some options?

**Comment by tomasfejfar** [comment]
@epdenouden certainly! I'd be happy to show you the use-case in detail and discuss how else we could approach that.

**Comment by Menrath** [comment]
I have a scenario where I also thought I might need this: A WordPress plugin that integrates with other plugins which might conflict. Splitting the integrations test up into a testsuite for each third-party-plugin-integration seemed the right way to go. But each testsuite requires a different bootstrap scenario to activate the corresponding third party plugins before the phpunit test suite is started. Coupling that to the testsuite name would be handy. Or maybe I just haven't learnt all the lessons about how to write good tests. My first approach was rather bad, but I want to improve :)

> I disagree that a test case should know the name of the test suite it is part of. [...]

@sebastianbergmann I agree, the test case should not know. But maybe the bootstrap should?
